security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -460,3 +476,40 @@ func (sc *SecretCache) isTokenExpired() bool {
 	// TODO(quanlin), check if token has expired.
 	return false
 }
+
+func constructCSRHostName(trustDomain, token string) (string, error) {
+	// If token is jwt format, construct host name from jwt with format like spiffe://cluster.local/ns/foo/sa/sleep,
+	strs := strings.Split(token, ".")
+	if len(strs) != 3 {
+		return "", fmt.Errorf("invalid k8s jwt token")
+	}
+
+	payload := strs[1]
+	if l := len(payload) % 4; l > 0 {
+		payload += strings.Repeat("=", 4-l)
+	}
+	dp, err := base64.URLEncoding.DecodeString(payload)
+	if err != nil {
+		return "", fmt.Errorf("invalid k8s jwt token")
```

Comment:
Please consider using a different error message, or put the error in the message.